source/conventions.md

```diff
@@ -171,23 +110,140 @@ lattice expansion where there are [`Fork`](#s:forking) elements and for evaluati
 expressions.
 
 The simplist form of name matching is if the string matches
-the `name` field of an element or elements. Regular expressions can be used. 
-Regular expressions must conform to the [PCRE2](https://www.pcre.org/) standard. 
+the `name` field of an element or elements. 
+For example, the string `"Q1"` will match to all elements named `Q1`.
 
-The names of an element may be "qualified" by prepending a `branch` or `BeamLine` name to the string
-using the string `">>"` as a separator. For example, `"B1>>Qaf.*"` would match
-to all elements in a branch or BeamLine named `"B1"` whose name begins with `"Qaf"`.
+Element matches may be restricted to a given element kind using the notation
+```{code} yaml
+<kind>::<name>
+```
+where `<kind>` is the element kind and `<name>` is the element name.
+Example:
+```{code} yaml
+Marker::bpm.
+```
+This will match to all `Marker` elements whose name is four characters starting with `bpm`.
+
+The `N`{sup}`th` element with a given name can be matched to by appending the character `"#"` 
+followed by an integer `N`. For example, `"Quadrupole::Q1#3"` will match to the third element
+that matches `"Quadrupole::Q1"`. Notice that the `N`{sup}`th` instance selection is applied last. 
 
-Additionally, if the match string ends with the character `"#"` followed by an integer `N`,
-this will match to the `N`{sup}`th` instance matched to in any `branch` or `BeamLine`.
-For example, `"Sd#3"` will match to the 3{sup}`rd` instance of all elements named
-`"Sd".
+<!--
+Besides the element name, any parameter in the [`MetaP`](#s:meta.params) parameter group can
+be matched to using the syntax `"{param-name}::{name-to-match}"`. For example, `"alias::black"`
+would match to any element whose `alias` parameter is set to `black`.
+-->
+
+In the discussion below, all of the above constructs will be called an "element name".
+
+The names of an element may be "qualified" by prepending a `branch` or `BeamLine` name 
+(henceforth just referred to as a branch name) to the string.
+using the string `">>"` as a separator. For example, `"B1>>Sextupole::Saf"` would match
+to all Sextupole elements in a branch or BeamLine named `"B1"` whose name was `"Saf"`.
+This includes sublines of BeamLines. Thus if `B1` is a BeamLine that contains a subline `B2` 
+that in turn contains a Sextupole element named `Saf`, the string `"B1>>Sextupole::Saf"`
+will match to this element.
 
 Lattice elements can also be referred to by the index in which they appear in a branch
-with the first element having index one, etc. Branches do not get an index since the
+with the first element having index one, etc. For example, `"B1>>7"` matches the seventh
```

Comment:
The other item we want to discuss in the meeting: shall PALS follow zero-based (C/C++/Python/etc.) or one-based (Fortran/Julia) indexing for arrays/lists?

Reply:
Seems like 1-based indexing was preferred, correct?

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/ff9c2e1f-4d38-4544-ae4d-178a1530876a" />

Not sure what conclusion to infer from the PCRE2 and CSS comments.